File: openapi-test-scala-server-http4s-client-http4s/testapi/api/PetsApiClient.scala
package testapi.api

import cats.effect.IO
import io.circe.Json
import java.lang.Void
import org.http4s.Uri
import org.http4s.client.Client
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate
import testapi.model.PetId
import org.http4s.circe.CirceEntityEncoder.circeEntityEncoder
import org.http4s.circe.CirceEntityDecoder.circeEntityDecoder

/** Http4s client implementation for PetsApi */
class PetsApiClient(
  /** Http4s client for making HTTP requests */
  val client: Client[IO],
  /** Base URI for API requests */
  val baseUri: Uri
) extends PetsApi {
  /** Create a pet */
  override def createPet(body: PetCreate): IO[Response201400[Pet, Error]] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.POST, baseUri / "pets").withEntity(body)
    client.run(request).use { response =>
        val statusCode = response.status.code
        if (statusCode == 201) response.as[testapi.model.Pet].map(v => testapi.api.Created(v))
        else if (statusCode == 400) response.as[testapi.model.Error].map(v => testapi.api.BadRequest(v))
        else cats.effect.IO.raiseError(new java.lang.IllegalStateException(s"Unexpected status code: ${statusCode}"))
    }
  }

  /** Delete a pet */
  override def deletePet(
    /** The pet ID */
    petId: PetId
  ): IO[Void] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.DELETE, baseUri / "pets" / petId)
    client.status(request).as(null.asInstanceOf[java.lang.Void])
  }

  /** Get a pet by ID */
  override def getPet(
    /** The pet ID */
    petId: PetId
  ): IO[Response200404[Pet, Error]] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.GET, baseUri / "pets" / petId)
    client.run(request).use { response =>
        val statusCode = response.status.code
        if (statusCode == 200) response.as[testapi.model.Pet].map(v => testapi.api.Ok(v))
        else if (statusCode == 404) response.as[testapi.model.Error].map(v => testapi.api.NotFound(v))
        else cats.effect.IO.raiseError(new java.lang.IllegalStateException(s"Unexpected status code: ${statusCode}"))
    }
  }

  /** Get pet photo */
  override def getPetPhoto(
    /** The pet ID */
    petId: PetId
  ): IO[Void] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.GET, baseUri / "pets" / petId / "photo")
    client.status(request).as(null.asInstanceOf[java.lang.Void])
  }

  /** List all pets */
  override def listPets(
    /** Maximum number of pets to return */
    limit: Option[Int],
    /** Filter by status */
    status: Option[String]
  ): IO[List[Pet]] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.GET, (baseUri / "pets").withOptionQueryParam("limit", limit).withOptionQueryParam("status", status))
    client.expect[scala.List[testapi.model.Pet]](request)
  }

  /** Upload a pet photo */
  override def uploadPetPhoto(
    /** The pet ID */
    petId: PetId,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array[Byte]
  ): IO[Json] = {
    val request = org.http4s.Request[cats.effect.IO](org.http4s.Method.POST, baseUri / "pets" / petId / "photo")
    client.expect[io.circe.Json](request)
  }
}

File: openapi-test-java-server-jaxrs-client-jdk/testapi/api/PetsApiClient.java
package testapi.api;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.lang.IllegalStateException;
import java.lang.Void;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpRequest.BodyPublishers;
import java.net.http.HttpResponse.BodyHandlers;
import java.util.List;
import java.util.Optional;
import testapi.model.Error;
import testapi.model.Pet;
import testapi.model.PetCreate;
import testapi.model.PetId;
import static typo.runtime.internal.stringInterpolator.str;

/** JDK HTTP Client implementation for PetsApi */
public class PetsApiClient implements PetsApi {
  HttpClient httpClient;;

  URI baseUri;;

  ObjectMapper objectMapper;;

  public PetsApiClient(
    /** JDK HTTP client for making HTTP requests */
    HttpClient httpClient,
    /** Base URI for API requests */
    URI baseUri,
    /** Jackson ObjectMapper for JSON serialization */
    ObjectMapper objectMapper
  ) {
    this.httpClient = httpClient;
    this.baseUri = baseUri;
    this.objectMapper = objectMapper;
  };

  /** Create a pet */
  @Override
  public Response201400<Pet, Error> createPet(PetCreate body) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets")).method("POST", BodyPublishers.ofString(objectMapper.writeValueAsString(body))).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    var statusCode = response.statusCode();
    if (statusCode == 201) { return new Created(objectMapper.readValue(response.body(), Pet.class)); }
    else if (statusCode == 400) { return new BadRequest(objectMapper.readValue(response.body(), Error.class)); }
    else { throw new IllegalStateException(str("Unexpected status code: ", java.lang.String.valueOf(statusCode), "")); }
  };

  /** Delete a pet */
  @Override
  public Void deletePet(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString())).method("DELETE", BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    return null;
  };

  /** Get a pet by ID */
  @Override
  public Response200404<Pet, Error> getPet(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString())).method("GET", BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    var statusCode = response.statusCode();
    if (statusCode == 200) { return new Ok(objectMapper.readValue(response.body(), Pet.class)); }
    else if (statusCode == 404) { return new NotFound(objectMapper.readValue(response.body(), Error.class)); }
    else { throw new IllegalStateException(str("Unexpected status code: ", java.lang.String.valueOf(statusCode), "")); }
  };

  /** Get pet photo */
  @Override
  public Void getPetPhoto(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString() + "/" + "photo")).method("GET", BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    return null;
  };

  /** List all pets */
  @Override
  public List<Pet> listPets(
    /** Maximum number of pets to return */
    Optional<Integer> limit,
    /** Filter by status */
    Optional<String> status
  ) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets" + (limit.isPresent() ? "?limit=" + limit.get().toString() : "") + (status.isPresent() ? "&status=" + status.get().toString() : ""))).method("GET", BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    return objectMapper.readValue(response.body(), new TypeReference<List<Pet>>() {});
  };

  /** Upload a pet photo */
  @Override
  public JsonNode uploadPetPhoto(
    /** The pet ID */
    PetId petId,
    /** Optional caption for the photo */
    String caption,
    /** The photo file to upload */
    Byte[] file
  ) throws java.lang.Exception {
    var request = HttpRequest.newBuilder(URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString() + "/" + "photo")).method("POST", BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build();
    var response = httpClient.send(request, BodyHandlers.ofString());
    return objectMapper.readValue(response.body(), JsonNode.class);
  };
}